AWSClientRuntime/Sources/Http/HttpResponse+AWS.swift

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+import ClientRuntime
+
+public extension HttpResponse {
+    /// Returns true if the status code is `HttpStatusCode.notFound` and the body is empty.
+    var statusCodeIsNotFoundAndBodyIsEmpty: Bool {
+        return statusCode == .notFound && body.isEmpty
+    }
+    
+    var requestId: String? {
+        return headers.value(for: "x-amz-request-id")
+    }
+    
+    var requestId2: String? {
+        return headers.value(for: "x-amz-id-2")
+    }
```

Comment:
I wonder if this extension should be `headers`  collection.

Reply:
Good question, and can go either way here. IMO the fact that these values come from headers are implementation details. That being said, it could be cool to live in a world where all the headers are defined on `Headers`.

Reply:
Makes sense.